documentation/commands.md

```diff
@@ -1,233 +1,243 @@
-                ["string_0_1_0", "string_0_1_1"],
-                ["string_0_2_0", "string_0_2_1"]
-              ],
-              [
-                ["string_1_0_0", "string_1_0_1"],
-                ["string_1_1_0", "string_1_1_1"],
-                ["string_1_2_0", "string_1_2_1"]
-              ]
-            ]
-          },
+          }
+        ],
+        "attributes": [
           {
-            "type": "dataset",
-            "name": "string_fixed_length_1d",
-            "dataset": {
-              "type":"string",
-              "string_size": 32,
-              "size": ["unlimited"]
-            },
-            "values": ["the-scalar-string", "another-one"],
-            "attributes": [
-            {
-              "name": "scalar_attribute",
-              "values": 42
-            },
-            {
-              "name": "vector_attribute",
-              "values": [1, 2, 3],
-              "type": "uint32"
-            }
-            ]
+            "name": "units",
+            "values": "ms"
           }
         ]
       }
     ]
-  },
-  "file_attributes": {
-    "file_name": "some.h5"
-  },
-  "cmd": "FileWriter_new",
-  "job_id": "unique-identifier",
-  "broker": "localhost:9092",
-  "start_time": "[OPTIONAL] timestamp (int) in milliseconds",
-  "stop_time": "[OPTIONAL] timestamp (int) in milliseconds",
-  "service_id": "[OPTIONAL] the_name_of_the_instance_which_should_interpret_this_command",
-  "abort_on_uninitialised_stream": false,
-  "use_hdf_swmr": true
 }
 ```
-By default, the file-writer will continue to write a file even when a stream fails to initialise.
-In order to abort writing instead `set abort_on_uninitialised_stream` to `true`.
 
-#### Command to exit the file-writer
 
-```json
-{
-  "cmd": "FileWriter_exit",
-  "service_id": "[OPTIONAL] the_name_of_the_instance_which_should_interpret_this_command"
-}
-```
+In the example above, a group called my_test_group is defined which in turn has a stream and an attribute defined.
+The stream definition instructs the file-writer that there is data available which should be written to the containing group.
+The parameters for the stream definition are:
 
-#### Command to stop a file being written
+- writer_module: The FlatBuffers schema in the [streaming-data-types](https://github.com/ess-dmsc/streaming-data-types) repository that was used to serialise the data.
+- type: The type of the data. The possible types are defined in the schema declared in the `writer_module`.
+- source: The name of the data source. For example, the EPICS PV name.
+- topic: The Kafka topic where the data can be found.
 
-```json
-{
-  "cmd": "FileWriter_stop",
-  "job_id": "job-unique-identifier",
-  "stop_time" : "[OPTIONAL] timestamp (int) in milliseconds",
-  "service_id": "[OPTIONAL] the_name_of_the_instance_which_should_interpret_this_command"
-}
-```
+Note: streams are automatically assigned a NeXus class based on the `writer_module`. [NEEDS TO BE CHECKED]
```

Comment:
No need to check, some writer modules automatically assigns NeXus class, some does not. See also [DM-1220](https://jira.esss.lu.se/browse/DM-1220?jql=project%20%3D%2016000%20AND%20labels%20%3D%20%22FW%26FW%22).

Reply:
There may even be multiple tickets about this because I think I created one on github at some stage.

Reply:
Remind me to nominate it for next time.